### PR TITLE
Fix redirected links to appropriate resolved URL references in Docs

### PR DESCRIPTION
**Pull request by nsenno-dbr**
Signed-off-by: nsenno-dbr <nick.senno@databricks.com>

## Related Issues/PRs

 #xxx

Updated links in CONTRIBUTION.md that had been renamed/removed 
Updated link to MLeap documentation that had been moved

(Please fill in changes proposed in this fix)

update to docs were tested by building the documents locally and ensuring links point to correct MLeap documentation

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests (describe details, including test results, below)

Validate docs rendering is correct and links function properly

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -522,7 +522,7 @@ the updated docs to your PR branch.
 ### Python
 
 If you are contributing in Python, make sure that you have satisfied all
-the [Common prerequisites and dependencies](), including installing
+the [Common prerequisites and dependencies](#environment-setup-and-python-configuration), including installing
```
Comment (BenWilson2) [inline CONTRIBUTING.md:525]:
```suggestion
the [Environment Setup and Python configuration](#environment-setup-and-python-configuration), including installing
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -503,7 +503,7 @@ and the Model Scoring Server for Java-based models like MLeap
 
 Other Java functionality (like artifact storage) depends on the Python
 package, so first install the Python package in a conda environment as
-described in [Common prerequisites and dependencies]().
+described in [Common prerequisites and dependencies](#environment-setup-and-python-configuration).
```
Comment (BenWilson2) [inline CONTRIBUTING.md:506]:
```suggestion
described in [Environment Setup and Python configuration](#environment-setup-and-python-configuration).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -786,7 +786,7 @@ python setup.py bdist_wheel
 ### Writing Docs
 
 First, install dependencies for building docs as described in [Common
-prerequisites and dependencies]().
+prerequisites and dependencies](#environment-setup-and-python-configuration).
```
Comment (BenWilson2) [inline CONTRIBUTING.md:789]: Same change here as above.

**Comment by nsenno-dbr** [comment]
@BenWilson2 updated CONTRIBUTION.md with your recommendations

**Review by BenWilson2 — APPROVED** [review]
LGTM! Thanks for the fixes @nsenno-dbr !

**Pull request merged**